incubator/patroni/templates/statefulset-patroni.yaml

```diff
@@ -3,91 +3,112 @@ kind: StatefulSet
 metadata:
   name: {{ template "patroni.fullname" . }}
   labels:
-    heritage: {{ .Release.Service | quote }}
-    release: {{ .Release.Name | quote }}
-    chart: "{{ .Chart.Name }}-{{ .Chart.Version }}"
-    component: "{{.Release.Name}}-{{.Values.Component}}"
+    app: {{ template "patroni.name" . }}
+    chart: {{ template "patroni.chart" . }}
+    release: {{ .Release.Name }}
+    heritage: {{ .Release.Service }}
 spec:
   serviceName: {{ template "patroni.fullname" . }}
-  replicas: {{default 5 .Values.Replicas }}
+  replicas: {{ .Values.replicaCount }}
+  selector:
+    matchLabels:
+      app: {{ template "patroni.name" . }}
+      release: {{ .Release.Name }}
   template:
     metadata:
       name: {{ template "patroni.fullname" . }}
       labels:
-        heritage: {{.Release.Service | quote }}
-        release: {{.Release.Name | quote }}
-        chart: "{{.Chart.Name}}-{{.Chart.Version}}"
-        component: "{{.Release.Name}}-{{.Values.Component}}"
+        app: {{ template "patroni.name" . }}
+        release: {{ .Release.Name }}
     spec:
-      {{ if .Values.NodeSelector }}
-      nodeSelector:
-        {{ range $key, $value := .Values.NodeSelector }}
-          {{ $key }}: {{ $value | quote }}
-        {{ end }}
-      {{ end }}
       serviceAccountName: {{ template "patroni.serviceAccountName" . }}
+      affinity:
```

Comment:
Please allow affinity to be configured the same way as e. g. node selectors or tolerations. This would be more flexible and would allow you to consider e. g. availability zones as well and not just hostnames.

Reply:
Added the option to set `affinity` rules, if you set these the `antiAffinity` will be ignored.

Reply:
I don't like this. The `podAntiAffinity` block is rather confusing. You should remove it. You have the full flexibilty to configure anything you need. You still may provide a default as follows using the `tpl` function.

https://github.com/kubernetes/charts/blob/494b80b66acdef9de90bcaf77eb1e5298ff9c0fc/incubator/vault/templates/deployment.yaml#L54
https://github.com/kubernetes/charts/blob/494b80b66acdef9de90bcaf77eb1e5298ff9c0fc/incubator/vault/values.yaml#L40-L49

Reply:
I wasn't aware of the `tpl` function, updated.